Sources/LemmyMarkdownUI/View/CodeBlockView.swift

```diff
@@ -22,15 +36,60 @@ internal struct CodeBlockView: View {
         }
         .background(configuration.codeBackgroundColor)
         .clipShape(RoundedRectangle(cornerRadius: 8))
+        .task(id: colorScheme) {
+            await performHighlighting()
+        }
+        .onDisappear {
+            highlightTask?.cancel()
+        }
     }
-    
+
+    @ViewBuilder
     var contentView: some View {
-        Text(content.trimmingCharacters(in: .newlines))
-            .font(
-                Font(
-                    configuration.font.withSize(configuration.font.pointSize * configuration.codeFontScaleFactor)
-                ).monospaced()
-            )
-            .padding(10)
+        if let highlighted = highlightedText,
+           configuration.enableSyntaxHighlighting {
+            Text(highlighted)
+                .font(codeFont)
+                .padding(10)
+        } else {
+            Text(content.trimmingCharacters(in: .newlines))
+                .font(codeFont)
+                .foregroundColor(configuration.primaryColor)
+                .padding(10)
+        }
+    }
+
+    var codeFont: Font {
+        Font(
+            configuration.font.withSize(configuration.font.pointSize * configuration.codeFontScaleFactor)
+        ).monospaced()
+    }
+
+    func performHighlighting() async {
+        guard configuration.enableSyntaxHighlighting else { return }
+        guard !isHighlighting else { return }
+
```

Comment:
At the moment, it sometimes tries to highlight code blocks that shouldn't be highlighted. E.g.

<img width=30% alt="Screenshot 2025-09-18 at 21 54 07" src="https://github.com/user-attachments/assets/ba09306e-3267-4c44-a215-1f344872ea3d" />

Maybe we could turn off highlighting for code blocks that don't have a language specified? Something like

```suggestion
    func performHighlighting() async {
        guard configuration.enableSyntaxHighlighting else { return }
        guard !isHighlighting else { return }
        guard language != nil else { return }

```

The downside of doing this is that it won't highlight code blocks where the user just forgot to add a language identifier. IMO that's better than accidentally highlighting things that shouldn't be highlighted, though.